enhancements/multi-arch/heterogeneous-architecture-clusters.md

```diff
@@ -0,0 +1,450 @@
+  to be imagestreams added for individual architectures and they would be picked based on the node architecture.
+  - references to some images like cluster-api-provider-aws, apiserver-network-proxy etc.. are x86. these would need to change to provide the images
+  for other architectures too. These images can be overridden through annotations today.
+- The nodepool object defaults to picking an x86 instancetype and AMI. The nodepool API does expose these fields so they can be changed manually to
+architecture appropriate values. The API has to make a provision to include the architecture value so the AMI and instance type can be chosen
+accordingly.
+
+### Console
+
+- As of today console filters content based on control plane architecture. This needs to be entirely removed or made configurable for
+heterogeneous clusters. Console should be able to filter out the set of OLM operators that, if installed will be able to find
+appropriate architecture nodes to run on. This means that it has to display only the subset of operators that match the architecture of the
+nodes on the cluster.
+- Console would provide a way for users to override nodeAffinity settings that would be set for each operator based on the architectures it
+supports.
+
+### Install
+
+For Phase 1, the way to add more workers to the cluster is to handcraft a machineset manifest file for a worker node and specify the architecture
+specific details like instancetype, image references and deploy the machineset to get a worker node. This can be pseudo automated further by creating
+the manifests using `openshift-install create manifests` and modifying or adding worker manifest files. Eventually the installer should support being
+able to declare differing compute and control plane sections in the install-config mainly allowing architectures to be different. In doing so it would
+also generate appropriate machineset and machineconfig files tailored to the specific architecture. The machineset file would contain the instancetype,
+image id (AMI or Azure disk image etc..) and region corresponding to the architecture chosen for the workers.
+
+### Open Questions
+- Would a cluster migration from homogeneous -> heterogeneous involve an upgrade or would the upgrade be a separate step after the migration?
+- How would CVO decipher the architecture of the cluster (homogeneous vs heterogeneous)? Would it have to parse the release image and extract the
+information from the image metadata?
+- How would Operators specify which architectures they support and how would OLM make this optionally configurable ? Is it through the nodeselectorterms
+as proposed or is there another mechanism?
+
+### Test Plan
+
+Testing for Phase 1:
+- QE would need to test heterogeneous nightly payloads before the entry is added to Cincinnati. This implies that the heterogeneous payloads will be
+added to the candidate channel a bit later than the homogeneous payloads.
```

Comment:
not clear why heterogeneous payloads would follow a different workflow from QE to Cincinnati or why this means they'd get to the channel later?

Reply:
this does not reflect the current plan. updated it